tests/unittests/test_custom_domain.py

```diff
@@ -27,3 +27,19 @@ def test_config_with_domain(self, mock_verify_access):
 
         # Verify domain in client is from config
         self.assertEqual(test_client.base_url, mock_config["base_url"])
+
+    def test_prepare_url(self, mock_verify_access):
+        """
+        Test if the correct params are added to url
+        """
+        mock_config = {'repository': 'singer-io/test-repo', "base_url": "http://CUSTOM-git.com", "access_token": "", "max_per_page": 35}
+        test_client = GithubClient(mock_config)
+
+        # Verify if per_page param was added as expected
+        self.assertEqual(test_client.prepare_url(test_client.base_url), f"{mock_config['base_url'].lower()}/?per_page=35")
+        self.assertEqual(test_client.prepare_url('http://CUSTOM-git.com/?q=query'), 'http://custom-git.com/?q=query&per_page=35')
+
+        # Verify if per_page param was added with default value
+        del mock_config["max_per_page"]
```

Comment:
Swap tests. Firstly test for default behaviour and then add `config['max_per_page'] = 35`